agent/cache/cache.go

```diff
@@ -554,9 +554,16 @@ func (c *Cache) fetch(key string, r getOptions, allowNew bool, attempt uint, ign
 		return ch
 	}
 
-	// If we already have an entry and it is actively fetching, then return
-	// the currently active waiter.
+	// If we already have an entry, it is actively fetching, and we don't need
+	// to revalidate the cache, then return the currently active waiter.
```

Comment:
at least to me, this comment doesn't make sense anymore since we say that `If we need to revalidate, ... ` below. Could we re-write it as ?

```suggestion
	// We have an entry that is actively fetching
	// If we need to revalidate, return with a closed channel
	// so the caller will immediately retry the cache.
	// Otherwise, return the currently active waiter.
```